docs/design/run.md

```diff
@@ -0,0 +1,275 @@
+conflict handling. These strategies then can be exposed in the ui with a 
+matching command.
+
+**Continue:** If any subprocess fails, we will continue the work on child 
+revisions. Notify the user on exit about the failed revisions. 
+
+**Stop:** Signal a fatal failure and cancel any scheduled work that has not
+yet started running, but let any already started subprocess finish. Notify the
+user about the failed command and display the generated error from the 
+subprocess. 
+
+**Fatal:** Signal a fatal failure and immediately stop processing and kill any 
+running processes. Notify the user that we failed to apply the command to the 
+specific revision. 
+
+We will leave any affected in its current state, if any subprocess fails. This
+allows us provide a better user experience, as leaving revisions in an 
+undesirable state, e.g partially formatted, may confuse users.
+
+## Resource constraints:
+
+It will be useful to constrain the execution to prevent resource exhaustion. 
+Relevant resources could include:
+- CPU and memory available on the machine running the commands. `jj run` can
+provide some simple mitigations like limiting parallelism to "number of CPUs" 
+by default, and limiting parallelism by dividing "available memory" by some 
+estimate or measurement of per-invocation memory use of the commands.
+- External resourec that are not immediately known to jj. For example, 
+commands run in parallel may wish to limit the toal number of connections
+to a server. We might choose to defer any handling of this to the 
+implementation of the command being invoked, instead of trying to 
+communicate that information to jj.
+
+
+## Command Options:
+
+The base command of any jj command should be usable. By default `jj run` works 
+on the `@` the current working copy.
+* --command, explicit name of the first argument
+* -x, for git compatibility (may alias another command)
+* -j, --jobs, the amount of parallelism to use
+* -k, --keep-going, continue on failure (may alias another command)
+* --show, display the diff for an affected revision
+* --dry-run, do the command execution without doing any work, logging all 
+intended files and arguments
+* --rebase, rebase all parents on the consulitng diff (may alias another 
+command)
+* --reparent, change the parent of an effected revision to the new change 
+(may alias another command)
+* --clean, remove existing workspaces and remove the ignored files
+* --readonly, ignore changes across multiple run invocations
+* --error-strategy=`continue|stop|fatal`, see [Dealing with failure](#Dealing-with-failure)
+
+### Integrating with other commands:
+
+`jj log`: No special handling needed
+`jj diff`: No special handling needed
+`jj st`: For now reprint the final output of `jj run`
+`jj op log`: No special handling needed, but awaits further discussion in 
+[#963][issue]
+`jj undo/jj op undo`: No special handling needed
+
+
+## Open Points:
+
+Should the command be backend specific?  
```

Comment:
Why would it be?

Reply:
Same question in the [doc](https://docs.google.com/document/d/14BiAoEEy_e-BRPHYpXRFjvHMfgYVKh-pKWzzTDi-v-g/edit?disco=AAAAmPkcYH0), can also be clarified.

Reply:
Yes, please clarify that this is not about commit backends (I think most people think of the commit backend when they hear an unqualified "backend").

Reply:
Adjusted to `working copy backend` in the second commit.